cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -296,7 +297,7 @@ struct page_total_size {
         }));
       auto const page_index =
         thrust::lower_bound(thrust::seq, iter + start, iter + end, i.row_index) - iter;
-      sum += c_info[page_index].size_bytes;
+      if (page_index < info_size) { sum += c_info[page_index].size_bytes; }
```

Comment:
If this is out of bound then that means the caller is wrongly executed on out-of-bound input. We should fix the caller below instead.

Reply:
I think it is a bit more complicated than that since the `page_index` is a result from searching through the `key_offsets` array and not a range value controlled by the kernel launch.